reasoner/planner/OrderingCoster.java

```diff
@@ -94,7 +94,7 @@ LocalAllCallsCosting createAllCallsCosting(ReasonerPlanner.CallMode callMode, Li
 
             estimator.extend(resolvable);
 
-            boolean isConnectedToInput = !Collections.intersection(resolvableMode, inputConnectedVars).isEmpty();
+            boolean isConnectedToInput = callMode.mode.isEmpty() || !Collections.intersection(resolvableMode, inputConnectedVars).isEmpty();
```

Comment:
I think it won't have had much of an effect, because the scaling factor in the case of empty bounds always 1 so the disconnected and acyclic costs should have equal weightage.